samples/databus/custom-serializer/Core_6/Shared/JsonDataBusSerializer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Text;
+
+namespace Shared
+{
+    using System.IO;
+    using Newtonsoft.Json;
+    using NServiceBus.DataBus;
+
+    #region CustomDataBusSerializer
+
+    public class JsonDataBusSerializer : IDataBusSerializer
+    {
+        public void Serialize(object databusProperty, Stream stream)
+        {
+            using (var writer = CreateNonClosingStreamWriter(stream))
+            using (var jsonWriter = new JsonTextWriter(writer))
+                serializer.Serialize(jsonWriter, databusProperty);
+        }
+
+        private StreamWriter CreateNonClosingStreamWriter(Stream stream)
+            => new StreamWriter(
+                stream,
+                Encoding.UTF8,
+                bufferSize: 1024,
+                leaveOpen: true);
+
+        public object Deserialize(Stream stream)
+        {
+            using (var reader = new StreamReader(stream))
+            using (var jsonReader = new JsonTextReader(reader))
+                return serializer.Deserialize(jsonReader);
+        }
+
+        private JsonSerializer serializer = JsonSerializer.Create(
```

Comment:
I don't know. The fact that this is a Json Serializer is kind of coincidental isn't it? The point is to show the NSB-facing API.